library/sha256.c

```diff
@@ -196,11 +196,26 @@ int mbedtls_internal_sha256_process( mbedtls_sha256_context *ctx,
         A[i] = ctx->state[i];
 
 #if defined(MBEDTLS_SHA256_SMALLER)
-    for( i = 0; i < 64; i++ )
     {
-        if( i < 16 )
+        uint32_t initval = mbedtls_platform_random_in_range(0xffffffff);
+        uint32_t offset = mbedtls_platform_random_in_range(16);
+        for( i = 0; i < 16; i++ )
+        {
+            W[i] = initval;
```

Comment:
Could we use `memset` or Mbed TLS version of it?

Reply:
Done